src/listpack.c

```diff
@@ -977,13 +977,15 @@ int lpValidateIntegrity(unsigned char *lp, size_t size, int deep,
     uint32_t count = 0;
     unsigned char *p = lpFirst(lp);
     while(p && p[0] != LP_EOF) {
```

Comment:
i'm still not sure about this change.
certainly your previous version who called the entry_cb after progressing to the `Next` element was wrong, and i do still think that the increment of `count++` should be last like before (not that it really makes any difference).

but regardless, i'm not sure about the added check of LP_EOF in the `while` loop.
in contrast to what you wrote, lpValidateNext does check for LP_EOF and sets the output var to NULL. but indeed that would not cause the loop to break, since it returns `1`. setting the output to NULL will cause the next iteration to break, but the `count` will still be incremented.
i think you're right that it should not be incremented, but then how come this check was passing (or how come your change didn't affect it)?
```c
    if (numele != LP_HDR_NUMELE_UNKNOWN && numele != count)
        return 0;
```

Reply:
Before this, ```deep``` of ```lpValidateIntegrity``` would never be set to 1, because this piece of code was dead, so the bug was never found.